src/plugins/condition/components/Criterion.vue

```diff
@@ -77,39 +89,78 @@ export default {
     data() {
         return {
             telemetryMetadata: {},
+            telemetryMetadataOptions: {},
             operations: OPERATIONS,
-            isInputOperation: false,
-            rowLabel: ''
+            inputCount: 0,
+            rowLabel: '',
+            operationFormat: ''
         }
     },
     computed: {
         setRowLabel: function () {
             let operator = this.trigger === 'all' ? 'and ': 'or ';
             return (this.index !== 0 ? operator : '') + 'when';
+        },
+        filteredOps: function () {
+            return [...this.operations.filter(op => op.appliesTo.indexOf(this.operationFormat) !== -1)];
         }
     },
     mounted() {
         this.updateMetadataOptions();
-        this.updateOperationInputVisibility();
     },
     methods: {
-        updateMetadataOptions() {
+        getOperationFormat() {
+            this.telemetryMetadata.valueMetadatas.forEach((value, index) => {
+                let valueMetadata = this.telemetryMetadataOptions[index];
+                if (valueMetadata.formatString) {
```

Comment:
Looks like SummaryWidgets did this differently: https://github.com/nasa/openmct/blob/master/src/plugins/summaryWidget/src/ConditionManager.js#L127
Can you check with @akhenry or @deeptailor about which approach is the right one?

Reply:
I like that implementation. I can follow the same pattern and make it a lot more readable.